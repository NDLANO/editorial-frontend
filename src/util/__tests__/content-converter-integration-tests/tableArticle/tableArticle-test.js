/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import jsdom from 'jsdom';
import prettify from '../prettify';
import {
  learningResourceContentToEditorValue,
  learningResourceContentToHTML,
} from '../../../articleContentConverter';
import { html } from './tableArticle';

const { fragment } = jsdom.JSDOM;

test('serializing article with table tag and attributes', () => {
  const converted = learningResourceContentToEditorValue(html, fragment);

  const result = learningResourceContentToHTML(converted);

  expect(prettify(result)).toMatchSnapshot();
});
