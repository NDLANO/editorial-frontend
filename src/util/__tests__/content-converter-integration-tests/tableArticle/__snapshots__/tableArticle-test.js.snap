// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializing article with sup tag 1`] = `
"
<section>
  <embed data-align
    data-alt=\\"Regnskap. Foto.\\"
    data-caption
    data-resource=\\"image\\"
    data-resource_id=\\"2935\\"
    data-size=\\"full\\"
    data-url=\\"https://api.ndla.no/image-api/v2/images/2935\\">

  <h2>Balanse</h2>
</section>
<section>
  <p>For å lære oss reglene skal vi se nærmere på balansen. Det er balansen som er selve «oppfinnelsen» når det gjelder måten vi fører regnskap på. Den danner grunnlaget for all regnskapsførsel. Måten balansen er bygd opp på, bestemmer hvordan alt innenfor regnskap skal føres.
  </p>
  <p>All kapital vi har skaffet oss, fører vi på høyre side under
    <em>Egenkapital og gjeld.</em>Vi kaller det gjerne
    <em>kapitalanskaffelsen
    </em>. Det vi har brukt kapitalen til, plasserer vi til venstre under
    <em>Eiendeler</em>. Det kaller vi
    <em>kapitalanvendelsen</em>, se illustrasjonen nedenfor:
  </p>
  <table>
    <thead>
      <tr>
        <th colspan=\\"4\\" align=\\"center\\" valign=\\"top\\">Balanse</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td colspan=\\"2\\"
          align=\\"center\\"
          valign=\\"top\\">
          <strong>Eiendeler</strong>
        </td>
        <td colspan=\\"2\\"
          align=\\"center\\"
          valign=\\"top\\">
          <strong>Egenkapital
            <br>
            og gjeld
          </strong>
        </td>
      </tr>
      <tr>
        <td colspan=\\"2\\"
          align=\\"left\\"
          valign=\\"top\\">Kapitalanvendelse:
          <br>
          hvor pengene er brukt
        </td>
        <td colspan=\\"2\\"
          align=\\"left\\"
          valign=\\"top\\">Kapitalanskaffelse:
          <br>
          hvor pengene kommer fra
        </td>
      </tr>
    </tbody>
  </table>
  <h2>Det doble bokføringsprinsippet</h2>
</section>
<section>
  <p>Dette uttrykket kan høres litt vanskelig ut. I balansen står på en måte de samme pengene på begge sider av oppstillingen. Det kaller vi
    <em>dobbel bokføring
    </em>. Bokføring er betegnelsen på å føre et regnskap. Fordi summen er lik på begge sider, snakker vi om balanse. For at regnskapet hele tida skal balansere, må vi alltid føre hendingene to ganger, en gang til høyre og en gang til venstre. Venstre side kaller vi
    <em>debet</em>, og høyre side er
    <em>kredit</em>.
  </p>
  <p>Når vi fører et bilag, må det alltid være
    <em>lik sum i debet og kredit.
    </em>Vi kan illustrere det med en såkalt T-konto, der vi for eksempel har solgt varer for 1000 kr kontant.
  </p>
  <table>
    <thead>
      <tr>
        <th colspan=\\"4\\" align=\\"center\\" valign=\\"top\\">Inntekter</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\">Debet</td>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\">Kredit</td>
      </tr>
      <tr>
        <td colspan=\\"2\\" align=\\"left\\" valign=\\"top\\"></td>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\">1 000</td>
      </tr>
    </tbody>
  </table>
  <table>
    <thead>
      <tr>
        <th colspan=\\"4\\" align=\\"center\\" valign=\\"top\\">Kontanter</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\">Debet</td>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\">Kredit</td>
      </tr>
      <tr>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\">1 000</td>
        <td colspan=\\"2\\" align=\\"center\\" valign=\\"top\\"></td>
      </tr>
    </tbody>
  </table>
  <p>Du ser at vi har ført samme sum en gang til kredit og en gang til debet. Hvorfor summen går i kredit på den ene kontoen og i debet på den andre kontoen, krever nærmere forklaring. Det er her tommelfingerreglene kommer inn.
  </p>
  <h2>Tre regnskapsregler</h2>
  <p>Balansen består av tre hovedgrupper: eiendeler, egenkapital og gjeld. Alle eiendelskontoene brukes på samme måte hver gang vi fører på dem. Det samme gjelder egenkapitalen og gjelda: Når vi fører i debet eller i kredit på en av kontoene, blir effekten alltid den samme.
  </p>
  <p>Det er lurt å tenke på hvor i balansen de tre hovedgruppene står:</p>
  <table>
    <thead>
      <tr>
        <th colspan=\\"4\\" align=\\"center\\" valign=\\"top\\">Balanse</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td colspan=\\"2\\"
          align=\\"center\\"
          valign=\\"top\\">
          <strong>Eiendeler</strong>
        </td>
        <td colspan=\\"2\\"
          align=\\"center\\"
          valign=\\"top\\">
          <strong>Egenkapital
            <br>
            og gjeld
          </strong>
        </td>
      </tr>
      <tr>
        <td valign=\\"top\\">Debet øker</td>
        <td valign=\\"top\\">Kredit minker</td>
        <td valign=\\"top\\">Debet minker</td>
        <td valign=\\"top\\">Kredit øker</td>
      </tr>
    </tbody>
  </table>
  <p>Ser du at de tre hovedgruppene oppfører seg etter hvor de står i balansen? Økning og minking skal føres samme vei som hovedgruppa står i balansen. Altså øker vi eiendelene ved å føre i debet, mens vi øker egenkapitalen og gjelda ved å føre i kredit.
  </p>
  <p>Vi kan derfor sette opp disse tre reglene:</p>
  <div class=\\"c-bodybox\\">
    <p>
      <strong>Regel 1</strong>
    </p>
    <p>Debitering av en eiendelskonto: øker verdien</p>
    <p>Kreditering av en eiendelskonto: reduserer verdien</p>
  </div>
  <div class=\\"c-bodybox\\">
    <p>
      <strong>Regel 2</strong>
    </p>
    <p>Debitering av en gjeldskonto: reduserer gjelda</p>
    <p>Kreditering av en gjeldskonto: øker gjelda</p>
  </div>
  <div class=\\"c-bodybox\\">
    <p>
      <strong>Regel 3</strong>
    </p>
    <p>Debitering av en egenkapitalkonto: reduserer egenkapitalen</p>
    <p>Kreditering av en egenkapitalkonto: øker egenkapitalen</p>
  </div>
  <p>Resultatregnskapet er en underavdeling av egenkapitalen. Derfor fungerer alle resultatkontoer som egenkapitalen. Egenkapitalen står til høyre i balansen (til kredit), og derfor fører vi en økning av egenkapitalen til høyre (til kredit) og en reduksjon til venstre (til debet).
  </p>
  <p>Slik er det med all føring i resultatregnskapet. Alt som øker egenkapitalen, blir ført i kredit. Alt som reduserer egenkapitalen, blir ført i debet. Det vil si at inntekter alltid skal føres i kredit, fordi det egentlig øker egenkapitalen. Kostnader skal alltid føres i debet, fordi det egentlig reduserer egenkapitalen.
  </p>
</section>"
`;
